router/internal/batch/batch_test.go

```diff
@@ -0,0 +1,147 @@
+package batch
+
+import (
+	"bytes"
+	"github.com/stretchr/testify/require"
+	"io"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestBatch(t *testing.T) {
+	t.Run("verify performance gains for huge strings", func(t *testing.T) {
+		rs := randomString(defaultBufioReaderSize*50, 'A')
+		fullString := `{"somestring": ` + rs + `}`
+
+		readerForFunc := bytes.NewBufferString(fullString)
+		funcStart := time.Now()
+		_, _, err := getFirstNonWhitespaceChar(readerForFunc, defaultBufioReaderSize)
+		funcElapsed := time.Since(funcStart)
+		require.NoError(t, err)
+
+		readAllReader := bytes.NewBufferString(fullString)
+		readAllStart := time.Now()
+		_, err = io.ReadAll(readAllReader)
+		readAllElapsed := time.Since(readAllStart)
+		require.NoError(t, err)
+
+		require.Greater(t, readAllElapsed, funcElapsed)
```

Comment:
Results of this test as follows
2025/04/08 15:34:04 readAllElapsed 647.25µs
2025/04/08 15:34:04 funcElapsed 4.25µs

It ofcourse only apply to large strings where the buffer is 2x or above the current buffer of 4kb

cc: @StarpTech